delivery_sendcloud_oca/models/delivery_carrier.py

```diff
@@ -0,0 +1,476 @@
+            order = self.env["sale.order"].browse(order_id)
+            # get sender address (warehouse)
+            warehouse = order.warehouse_id
+            if not warehouse.sencloud_sender_address_id:
+                # use standard server address
+                sender_address = self._get_default_sender_address_per_company(
+                    warehouse.company_id.id
+                )
+            else:
+                sender_address = warehouse.sencloud_sender_address_id
+
+            # filter by weight
+            # TODO are there sendcloud shipping methods without weight limit?
+            weight = order.sendcloud_order_weight
+            sendcloud_carriers = sendcloud_carriers.filtered(
+                lambda c: c.sendcloud_min_weight <= weight <= c.sendcloud_max_weight
+            )
+
+            # filter by sender address (warehouse) and delivery address (partner)
+            countries = self.env["sendcloud.shipping.method.country"].search(
+                [
+                    ("company_id", "=", order.company_id.id),
+                    ("from_iso_2", "=", sender_address.country),
+                    ("iso_2", "=", partner.country_id.code),
+                ]
+            )
+            method_codes = countries.mapped("method_code")
+            sendcloud_carriers = sendcloud_carriers.filtered(
+                lambda c: c.sendcloud_code in method_codes
+            )
+
+            # filter out carriers requiring service points that are not enabled
+            without_service_point = sendcloud_carriers.filtered(
+                lambda c: c.sendcloud_service_point_input != "required"
+            )
+            with_service_point = sendcloud_carriers.filtered(
+                lambda c: c.sendcloud_service_point_input == "required"
+                and c.sendcloud_integration_id.service_point_enabled
+            )
+            enabled_service_point = self.env["delivery.carrier"]
+            for carrier in with_service_point:
+                carrier_names = carrier.sendcloud_integration_id.service_point_carriers
+                current_carrier = carrier.sendcloud_carrier
+                if (
+                    current_carrier
+                    and current_carrier in safe_eval(carrier_names)
+                    or []
+                ):
+                    enabled_service_point += carrier
+            sendcloud_carriers = without_service_point + enabled_service_point
+
+        return (sendcloud_carriers + other_carriers).sorted(
+            key=lambda carrier: carrier.name
+        )
+
+    # ----------------- #
+    # Sendcloud methods #
+    # ----------------- #
+
+    def _sendcloud_set_countries(self):
+        for record in self.filtered(
+            lambda r: r.delivery_type == "sendcloud" and r.sendcloud_sync_countries
+        ):
+            record.country_ids = record._sendcloud_get_countries_from_cache()
+
+    def _sendcloud_get_countries_from_cache(self):
```

Comment:
Maybe the method name deos not reflect the current behavior.

Reply:
Changed the method name